src/target/riscv/batch.c

```diff
@@ -34,6 +34,7 @@ struct riscv_batch *riscv_batch_alloc(struct target *target, size_t scans)
 	out->data_out = NULL;
 	out->data_in = NULL;
 	out->fields = NULL;
+	out->delay_types = NULL;
```

Comment:
Detail - optional:

I believe `delay class` is somewhat more descriptive than `delay type`.

```suggestion
	out->delay_classes = NULL;
```

Reply:
Addressed.